src/robusta/core/discovery/discovery.py

```diff
@@ -70,87 +72,132 @@ def discovery_process() -> DiscoveryResults:
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                active_services.extend(
+                    [
+                        Discovery.__create_service_info(
+                            replicaset.metadata,
+                            "ReplicaSet",
+                            extract_containers(replicaset),
+                            extract_volumes(replicaset),
+                            extract_total_pods(replicaset),
+                            extract_ready_pods(replicaset),
+                        )
+                        for replicaset in replicasets.items
+                        if not replicaset.metadata.owner_references and replicaset.spec.replicas > 0
+                    ]
+                )
+                continue_ref = replicasets.metadata._continue
+                if not continue_ref:
+                    break
+
+            # discover pods
+            continue_ref = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                pods: V1PodList = client.CoreV1Api().list_pod_for_all_namespaces(
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                pod_items = pods.items
+                active_services.extend(
+                    [
+                        Discovery.__create_service_info(
+                            pod.metadata,
+                            "Pod",
+                            extract_containers(pod),
+                            extract_volumes(pod),
+                            extract_total_pods(pod),
+                            extract_ready_pods(pod),
+                        )
+                        for pod in pod_items
+                        if not pod.metadata.owner_references and not is_pod_finished(pod)
+                    ]
+                )
+                continue_ref = pods.metadata._continue
+                if not continue_ref:
+                    break
 
-            pods: V1PodList = client.CoreV1Api().list_pod_for_all_namespaces()
-            pod_items = pods.items
-            active_services.extend(
-                [
-                    Discovery.__create_service_info(
-                        pod.metadata,
-                        "Pod",
-                        extract_containers(pod),
-                        extract_volumes(pod),
-                        extract_total_pods(pod),
-                        extract_ready_pods(pod),
-                    )
-                    for pod in pod_items
-                    if not pod.metadata.owner_references and not is_pod_finished(pod)
-                ]
-            )
         except Exception:
             logging.error(
                 "Failed to run periodic service discovery",
                 exc_info=True,
             )
 
-        # discover nodes
+        # discover nodes - no need for batching. Number of nodes is not big enough
```

Comment:
Bug-
nodes uses `pod_items` for getting the pods list so it will need batching too.
the pod_items here will only be the last batch run in the pods discovery so the rest of the pods will be missing in discovery

Reply:
fixed